src/usermod.c

```diff
@@ -804,7 +804,7 @@ static void update_gshadow (void)
 	bool was_admin;
 	bool changed;
 	const struct sgrp *sgrp;
-	struct sgrp *nsgrp;
+	struct sgrp *nsgrp = NULL;
 
 	changed = false;
 
```

Comment:
I honestly don't like this patch.  Initializing stuff is usually bad, since it disables the ability of static analyzers to diagnose uninitialized uses.  It also makes it more complex to handle it correctly, which in this specific case can be seen by the need to have two free(3) calls, and set to NULL after free(3).

Regarding your feedback about using goto: That use of goto in my patch is just forward jumping to right before the continue point, so it doesn't do any magic.

If you still don't want to use goto in the loop, I can prepare some other refactor, adding a helper function.

I'll reject this patch, and prepare some alternative patch in a new PR.

Reply:
Dropped in favour of https://github.com/shadow-maint/shadow/pull/996